### PR TITLE
call AddData with DemuxPacket

### PR DESCRIPTION
**Pull request by peak3d**
see title. only fixed the places I found, most likely I missed something....

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp [hunk 1]
```diff
@@ -137,7 +137,7 @@ void CDVDAudioCodecFFmpeg::Dispose()
   avcodec_free_context(&m_pCodecContext);
 }
 
-int CDVDAudioCodecFFmpeg::AddData(uint8_t* pData, int iSize, double dts, double pts)
+int CDVDAudioCodecFFmpeg::AddData(DemuxPacket &data)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: maybe call in pkt, or packet?

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxPacket.h [hunk 2]
```diff
@@ -19,6 +19,7 @@
  *  <http://www.gnu.org/licenses/>.
  *
  */
+#include "cores/VideoPlayer/DVDClock.h"
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxPacket.h]: what is that?

Comment (peak3d) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxPacket.h]: DVD_NO_PTS

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxPacket.h]: better pass those as parameters to constructor, like pData and iSize. lower layers should not include headers from upper layers.

**Comment by peak3d** [comment]
things should be OK now.....

File: xbmc/cores/VideoPlayer/CMakeLists.txt [hunk 3]
```diff
@@ -17,6 +17,7 @@ set(SOURCES DVDAudio.cpp
 
 set(HEADERS DVDAudio.h
             DVDClock.h
+            DVDConstants.h
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/CMakeLists.txt]: I merge this now but please rename this. "DVD" is history

**Pull request merged**